bson/src/main/org/bson/codecs/pojo/RecordCodecProvider.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.codecs.pojo;
+
+import org.bson.codecs.Codec;
+import org.bson.codecs.configuration.CodecProvider;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.diagnostics.Logger;
+import org.bson.diagnostics.Loggers;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static org.bson.assertions.Assertions.notNull;
+
+/**
+ * Provides Codecs for registered Records via the ClassModel abstractions.
+ *
+ * @since TODO
+ */
+public final class RecordCodecProvider implements CodecProvider {
```

Comment:
I see its to do with the PojoBuilderHelper

Reply:
It does differ a little (because of PojoBuilderHelper like you said), but you're right that we don't need a separate provider. My motivation for doing it this way was so that users could decide whether they wanted to support just records, just Pojos, or both. Now I'm thinking that we can choose a reasonable default for that (probably support both), and add a configuration in PjoCodecProvider where a user can specify if they want something different. Is that's even a useful option or should we just force users to support both?

Reply:
Unlike POJO codec, I think record codec can be enabled safely by default.  That was my motivation for having a separate provider for records.

Reply:
@rozza what do you think of what Jeff said? One option is we could just use PojoCodecProvider and by default have it support records, and require it to be manually configured if you want to support pojos as well.

Reply:
I'm happy to defer this decision for now.  I think we'll just have to consider if users will use Records in isolation to Pojo's or if they will combine them.  As its such a new feature we just don't have that data yet.  

@jyemin 's idea of having on by default Record support would be really good.